tests/test_pr_comments.py

```diff
@@ -215,6 +220,18 @@ def test_has_contractor_comment_no_comments(app, reqctx, requests_mocker):
     assert result is False
 
 
+def test_has_internal_cherry_pick_comment_release_environ_set(app):
+    pr = make_pull_request(user='nedbat')
+    open_edx_release = "Matt's Open edX Release"
+    fun_facts = get_fun_fact_file()
+    question, answer = random.choice(list(fun_facts.items()))
+    with app.test_request_context('/'):
+        comment = github_internal_cherrypick_comment(pr, open_edx_release, question, answer)
+    assert "Matt's Open edX Release" in comment
+    assert "Which animal's eye" in comment
+    assert "hide their head in the sand" in comment
+
+
```

Comment:
You need a test of the logic in `pull_request_opened`, and in `github_prepare_cherry_pick_comment`.